packages/report-server/src/routes/FetchReportRoute.ts

```diff
@@ -32,36 +32,63 @@ export class FetchReportRoute extends Route<FetchReportRequest> {
     return report;
   }
 
-  async checkUserHasAccessToReport(
-    report: ReportType,
+  async getRequestedOrgUnitObjects(
     hierarchy = 'explore',
     requestedOrgUnitCodes: string[],
-  ) {
-    const { accessPolicy, ctx } = this.req;
-    const permissionGroupName = await report.permissionGroupName();
-
-    const foundOrgUnits = await ctx.microServices.entityApi.getEntities(
+  ): Promise<{ code: string; country_code: string | null; type: string | null }[]> {
```

Comment:
```suggestion
  ) {
```
I know I previously suggested we still have a return type listed here, but on second thought I feel it may be best just to leave it up to typescript to do this automatically. Typescript is quite smart like that, it basically looks at all the `return` statements of a function, and infers that the function return type is just a union type of all the individual return types. I think using this automatic completion is ideal when we aren't too concerned with the explicit return type of the function 👍

Reply:
Thanks Rohan, that's something that I didn't know of. Yeah that is pretty smart!